movement/filesystem.c

```diff
@@ -166,6 +166,28 @@ bool filesystem_read_file(char *filename, char *buf, int32_t length) {
     return false;
 }
 
+bool filesystem_read_line(char *filename, char *buf, int32_t offset, int32_t length) {
+    memset(buf, 0, length);
+    int32_t file_size = filesystem_get_file_size(filename);
+    if (file_size > 0) {
+        int err = lfs_file_open(&lfs, &file, filename, LFS_O_RDONLY);
+        if (err < 0) return false;
+        err = lfs_file_seek(&lfs, &file, offset, LFS_SEEK_SET);
+        if (err < 0) return false;
+        err = lfs_file_read(&lfs, &file, buf, min(length, file_size - offset));
```

Comment:
Yep, this seems like the move :)

Reply:
Instead of reading to length - 1, I clarified the documentation to state that you should provide a buffer of length + 1, and that the last character will be a null terminator; this is how we were currently using the function anyway, and it feels more obvious than reading one character less than we were asked for.